packages/fxa-auth-server/lib/routes/password.ts

```diff
@@ -595,6 +596,394 @@ module.exports = function (
         }
       },
     },
+    {
+      method: 'POST',
+      path: '/mfa/password/change',
+      options: {
+        ...PASSWORD_DOCS.PASSWORD_CHANGE_JWT_POST,
+        auth: {
+          strategy: 'mfa',
+          scope: ['mfa:password'],
+          payload: false,
+        },
+        validate: {
+          query: isA.object({
+            keys: isA.boolean().optional().description(DESCRIPTION.queryKeys),
+          }),
+          payload: isA
+            .object({
+              email: validators.email().description(DESCRIPTION.email),
+              oldAuthPW: validators.authPW.description(DESCRIPTION.authPW),
+              authPW: validators.authPW.description(DESCRIPTION.authPW),
+              authPWVersion2: validators.authPW
+                .optional()
+                .description(DESCRIPTION.authPW),
+              wrapKb: validators.wrapKb
+                .optional()
+                .description(DESCRIPTION.wrapKb),
+              wrapKbVersion2: validators.wrapKb
+                .optional()
+                .description(DESCRIPTION.wrapKbVersion2),
+              clientSalt: validators.clientSalt
+                .optional()
+                .description(DESCRIPTION.clientSalt),
+            })
+            .and('authPWVersion2', 'wrapKbVersion2', 'clientSalt'),
+        },
+      },
+      handler: async function (request: AuthRequest) {
+        log.begin('Password.changeJWT', request);
+        const sessionToken = request.auth.credentials as any;
+        const uid = sessionToken.uid;
+
+        // MFA strategy already fetched and validated the session token
+        // The credential object contains the session token data
+        const {
```

Comment:
Instead of copying can we just call into the old password change routine? Sorta like this, https://github.com/mozilla/fxa/pull/19444#discussion_r2349755280.